Wanderer translator instead of Hai translator (#16)
UnorderedSigh/endless-sky: data/wanderer/wanderers start.txt

		accept

mission "Human Cultural Archives"
	description "Eruk, the Hai technologist on <origin>, has agreed to give you a translation machine if you can somehow get him a copy of the human cultural data from <destination>."
mission "Wanderers: Translation Algorithm Training Data"
	description "Bring a copy of the human cultural archives to <destination> so the Wanderers can train an automated translation algorithm."
	landing
	source "Greenwater"
	destination "Alexandria"
	source "Alexandria"
	destination "Vara K'chrai"
	to offer
		has "Wanderers: Translation Machine: done"
		not "human cultural data"
	on offer
		conversation "eruk cultural data"
	on visit
		dialog
			`The station has a small museum gift shop that sells a copy of the entire archive on a data card. It only costs 40 credits, but because you have done a horrible job at managing your finances you do not have even that much cash on hand right now.`

> 			`	Go earn some money, then return here.`

Reviewer: I really want to keep this scathing telling-off of the player.

Author: This is actually the critical `on visit` that I removed. Thank you for noticing that.

Author: Now I've noticed the old "on visit" text makes no sense, and the "on complete" makes no sense if you saw the "on visit" first. I will have to update both in a few hours once I have time.

EDIT: Also, the "on visit" doesn't work in its current location. I will need to move the "on offer" to the prior mission as an "on complete."

Author: It took some doing, but I got it to work and tested both routes:

1. If you lack 40 credits, the visit message has the library search and scolds you for being bankrupt. Completion message congratulates you for paying 40 credits, and does not do the search.
2. If you have 40 credits, the completion message has the library search and you pay 40 credits to finish.

> 			`You travel through the library searching for data suitable for training the Wanderer translation algorithm but you have an unexpected problem: you're surrounded by it. Even a Bulk Freighter would be unable to carry all the tomes and data cores of Alexandria library. Someone with expertise in natural language processing would know what to do. Asking around, you're referred to a research librarian who develops automated translation algorithms for dead languages. She directs you to the museum gift shop, where you buy the "Human Cultural Archive" for 40 credits.`

Reviewer: ```suggestion
			`You travel through the library searching for data suitable for training the Wanderer translation algorithm, but you have an unexpected problem: you're surrounded by it. Even fifty Bulk Freighters would be unable to carry all the tomes and data cores of Alexandria library. Someone with expertise in natural language processing would know what to do. Asking around, you're referred to a research librarian who develops automated translation algorithms for dead languages. She directs you to the museum gift shop, where you buy the "Human Cultural Archive" for 40 credits.`
```
Alexandria must be the British Museum, the Smithsonian, the British Library (which has 170 million books by itself), the Library of Congress, and every other national and minor library and museum on Earth, Chiron, Tinker, Reunion, Mars, Hestia, Martini, Zug, Moonshake, Solace, Bourne, Hephaestus, put together... there must be at least fifty Bulk Freighters for all of humanity's cultural legacy and production in physical form.

All of that fits on a human data card for fourty credits, hahaha. Quantum computing in 1000 years...

				accept
	on accept
		"reputation: Wanderer" >?= 10
	on complete
		set "human cultural data"
		payment -40
		conversation
			`<Library search> She directs you to the museum gift shop, where you buy the "Human Cultural Archive" for 40 credits.`
				to display
					not "Bankrupt at Alexandria"
			`Now that you finally have 40 credits, you purchase the "Human Cultural Archive" that you must take to <planet>.`
				to display
					has "Bankrupt at Alexandria"
	on visit
		set "Bankrupt at Alexandria"
		conversation
			`<Library search> She directs you to the museum gift shop, to buy the "Human Cultural Archive" for 40 credits.`
			`	Unfortunately, you have done such a horrible job at managing your finances you do not have even that much cash on hand right now.`
			`	Go earn some money, then return here.`

mission "Wanderers: Algorithm Training Data"
	description "Bring a copy of the human cultural archives to <destination> so the Wanderers can train an automated translation algorithm."
	landing
	source "Alexandria"
	destination "Vara K'chrai"
	to offer
		has "Wanderers: Translation Machine: done"
	on complete
		event "wanderer translator available" 84
		conversation

> 			`A Wanderer meets you at landing and you hand them the data card. They plug it into a mango-sized machine whose data card slot seems to adjust itself to match the shape of the card on the way in. The Wanderer makes clicking and whistling noises while manipulating the device. Eventually, the device presents the words, "[Enter, Bring] ten and two. Within [truth, lies], the pleasant [cow, nova] tickles me [thoughtfully, gracefully]."`

Reviewer 2: ```suggestion
			`A Wanderer meets you at landing and you hand them the data card. They plug it into a mango-sized machine whose data card slot seems to adjust itself to match the shape of the card on the way in. The Wanderer makes clicking and whistling noises while manipulating the device. Eventually, the device presents the words, "[Enter, bring] ten and two. Within [truth, lies], the pleasant [cow, nova] tickles me [thoughtfully, gracefully]."`
```

> 			`	The Wanderer seems displeased with the results. After a few minutes of fiddling with the device, it sputters out, "Pleasing [truth, gratifications]. [Accepting, Holding] geolocated results [within, for] three [menstrual, lunar] cycles." Seeing your confusion, the Wanderer manipulates the device for several more minutes.`

Reviewer 2: ```suggestion
			`	The Wanderer seems displeased with the results. After a few minutes of fiddling with the device, it sputters out, "Pleasing [truth, gratifications]. [Accepting, holding] geolocated results [within, for] three [menstrual, lunar] cycles." Seeing your confusion, the Wanderer manipulates the device for several more minutes.`
```
![image](https://user-images.githubusercontent.com/4471575/222343674-68544f27-0659-4675-8e89-1d686461a8de.png)

Looks like the standard for ambiguous translations at the start of a sentence is to only capitalize the first term.

# Compatibility patch for saves before "language: Hai" variable was added.

> mission "Hai Language Patch"

Reviewer: Are there other missions that depend on you having the "language: Hai" condition? Best to check.

Author: There are none that require it, but there are some with alternative dialog. I will add a Hai translator in another PR.

Author: That'll be a single mission, and quite a short one, so I'll PR it directly to endless-sky master. It's just going to say that the ban on human-Hai translators has been lifted, so the translators that have been available for a century are allowed now. They're government-subsidized, so you'll be offered one for free if you land on any inhabited Hai planet or station.

Author: It is in my fork's `hai-translator-ban-lifted` branch, and I'll PR it at the same time as this branch.

			`It is possible that the worlds they currently inhabit were formerly part of Hai territory. That might explain why the Unfettered believe they have a right to take that territory back.`
		log "People" "Iktat Rek" `Rek is a Wanderer ambassador, a new and unusual profession among a species that until recently has had no contact with aliens except for the occasional Unfettered Hai raid. His job is to help the Wanderers develop a deeper relationship with humans.`
		conversation
			`After you land, a group of Wanderers take you to the engineer from your last meeting. Explaining the device, the engineer says, "It will [function, succeed] on speech of typical [speed, tempo], but-" As they speak faster, the translation device produces gibberish. "This is Iktat Rek," they say, gesturing to another Wanderer. "He is [appointed, chosen] as the official [envoy, bearer] for human [visitors, couriers]."`

> 			`	Rek is slightly larger than most of the Wanderers you have seen, with muted white and brown plumage and a calm, dignified bearing. "It is very [rare, exceptional] in our history that we have to [deal with, cope with] other species," he says, "and now we find ourselves dealing with two at once."`

Reviewer: ```suggestion
			`	Rek is slightly larger than most of the Wanderers you have seen, with muted white and brown plumage and a calm, dignified bearing. "It is very [rare, exceptional] in our history that we have to [deal with, cope with] other species," he says, "and now we find ourselves [dealing with, coping with] two at once."`
```
Is it intentional that the device doesn't repeat itself again?

Author: I don't know. That was part of the original. Maybe MZ thought the dialog would be too repetitive that way?

I'll put in your change and see what happens in review.

			`	"[Automated, machine] language translation is a [primitive, early] application of machine [learning, intelligence]. For us it is [elementary, instinctual] mathematics. Our [thinkers, tinkerers] told me that most of the [work, enjoyment] was adapting the [interface, conceptualization, mechanism] to a human mind."`
			choice
				`	"How did you learn the Hai language?"`
					goto language
				`	"How long have you been fighting the Unfettered Hai?"`
					goto fighting

			label fighting
			`	"It has been nearly twenty [years, cycles] now," he says. "At first only a single Hai ship [visited, invaded] us, but back then we had no defenses in place to stop it. For years the ship only [harvested, stole] food from our frontier worlds, and avoided combat. In our culture it is an [obligation, duty] to share our food with one who [asks for, demands] it, so we did not oppose the Hai ship. But then more Hai ships came, attacked one of our worlds and took it for themselves, and killed many of our people. We built warships and drove them back, and yet their [raids, attacks] still continue."`
				goto food

> 			

Reviewer: label `language` may be worth changing. Since they have those algorithms, you could mention that some Wanderers have learned the Hai language with the help of those algorithms and devices.

Author: You need quite a substantial amount of data to train a language translation program. Also, every race in the galaxy that has had computers for more than a century or so will have these algorithms; there's nothing special about the Wanderers having them.

Author: The Wanderers explain this now. Actually, it is a critical plot point because it explains why they don't have a Hai translator. Thank you for finding this.

			branch hint
				has "Wanderers: Defend Vara Ke'sok Hint: done"
			`Soon after you land in the spaceport, there is a flurry of activity as many of the Wanderers fly to their ships, which begin to take off from the planet. In the sky, you see bright flashes that may be distant weapons fire or explosions.`

> 			`	You are buffeted by wind as one of the Wanderers lands next to your ship and says, in their language, "Unfettered [raider, pirate] fleet comes. We have put food on freighter, hope they take [offering, gift] and leave. If not [appeased, satisfied] you help convince them to leave. Yes?"`

Reviewer: ```suggestion
			`	You are buffeted by wind as one of the Wanderers lands next to your ship and says, "Unfettered [raider, pirate] fleet comes. We have put food on freighter, hope they take [offering, gift] and leave. If not [appeased, satisfied] you help convince them to leave. Yes?"`
```
It might be easier to just get rid of that phrase.

			branch hint
				has "Wanderers: Defend Vara Ke'sok Hint: done"
			`Soon after you land in the spaceport, there is a flurry of activity as many of the Wanderers fly to their ships, which begin to take off from the planet. In the sky, you see bright flashes that may be distant weapons fire or explosions.`

> 			`	You are buffeted by wind as one of the Wanderers lands next to your ship and says, "An Unfettered [raider, pirate] fleet comes. We have put food on a freighter, hoping they take the [offering, gift] and leave. If they're not [appeased, satisfied], will you [compel, influence] them to leave?"`

Reviewer: I thought about this, but I ignored it because I thought the Wanderer, while in flight, was trying to say as little as possible to convey you the information. It was just a little bit of variation.

Author: My interpretation is that these Wanderers were less skilled at speaking Hai than the ambassadors and scientists, and the translator correctly translated their grammatically-incorrect sentences.

Author: I have reverted the text and added this: `Their words are so quick, your translator can barely keep up.`
